### DISCUSSION TITLE
When to invoke hid_close?

### DISCUSSION THREAD

**Discussion by gfisys** [libusb/hidapi#257] [Q&A] [answered]
In my app, usb transfer occur frequently, carrying only small payload, 4-8 bytes at a time. Should I call hid_close at the end of every transfer (and call hid_open at the beginning of every transfer)? Or I can do so only when the app exits? Only a single thread will be accessing usb at any time.

Henry

**Comment by Youw** [comment, marked as answer]
When the app exits
